### PR TITLE
Add delete deployment cli

### PR DESCRIPTION
**Pull request by ColeMurray**
## Summary

Add Delete Deployment to CLI

To allow developers/admin to delete deployments from CLI, add command to cli and methods to api client.

## Changes

Adds functionality to CLI to delete a deployment by deployment id

## Importance

Orion developers currently do not have an easy way to delete deployments from CLI

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

File: src/prefect/cli/deployment.py [hunk 1]
```diff
@@ -262,3 +262,22 @@ async def create(path: str):
...
+
+@deployment_app.command()
+async def delete(deployment_id: str):
+    """
+    Delete a deployment.
+
+    \b
+    Example:
+        \b
+        $ prefect deployment inspect dfd3e220-a130-4149-9af6-8d487e02fea6
```
Comment (anna-geller) [inline src/prefect/cli/deployment.py:275]: this should be `prefect deployment delete UUID`

File: src/prefect/cli/deployment.py [hunk 2]
```diff
@@ -262,3 +262,22 @@ async def create(path: str):
                 "View your new deployment with: "
                 f"\n\n    prefect deployment inspect {stylized_name}"
             )
+
+
+@deployment_app.command()
+async def delete(deployment_id: str):
```
Comment (madkinsz) [inline src/prefect/cli/deployment.py:268]:
```suggestion
async def delete(deployment_id: UUID):
```

File: src/prefect/cli/deployment.py [hunk 3]
```diff
@@ -262,3 +262,22 @@ async def create(path: str):
...
+        \b
+        $ prefect deployment delete dfd3e220-a130-4149-9af6-8d487e02fea6
+    """
+    async with get_client() as client:
+        try:
+            deployment = await client.delete_deployment(deployment_id)
+        except ObjectNotFound:
+            exit_with_error(f"Deployment {deployment_id!r} not found!")
+
+    console.print(f"Deleted deployment {deployment_id}")
```
Comment (madkinsz) [inline src/prefect/cli/deployment.py:283]:
```suggestion
    exit_with_success(f"Deleted deployment '{deployment_id}'.")
```

File: src/prefect/cli/deployment.py [hunk 4]
```diff
@@ -262,3 +262,22 @@ async def create(path: str):
...
+    \b
+    Example:
+        \b
+        $ prefect deployment delete dfd3e220-a130-4149-9af6-8d487e02fea6
+    """
+    async with get_client() as client:
+        try:
+            deployment = await client.delete_deployment(deployment_id)
+        except ObjectNotFound:
+            exit_with_error(f"Deployment {deployment_id!r} not found!")
```
Comment (madkinsz) [inline src/prefect/cli/deployment.py:281]: We can't use repr for UUIDs and I've suggested updating the input type

```suggestion
            exit_with_error(f"Deployment '{deployment_id}' not found!")
```

**Review by madkinsz — COMMENTED** [review]
Thanks! Some minor requested changes.

**Comment by madkinsz** [comment]
The 3.7 failure looks unrelated, but the UUID type is not defined in the other failures. You'll need to import it.

**Pull request merged**